packages/lambda-shared/src/aws/tile.metadata.table.ts

```diff
@@ -111,29 +120,30 @@ export class TileMetadataTable {
      * @param Keys Imagery ids (already prefixed `im_${key}`)
      */
     private async fetchImagery(keys: string[]): Promise<TileMetadataImageryRecord[]> {
-        // TODO run multiple requests
-        if (keys.length > 100) throw new Error('BatchGet only allows 100 keys');
-
-        const Keys = keys.map(toId);
-        const items = await this.dynamo
-            .batchGetItem({
-                RequestItems: {
-                    [Const.TileMetadata.TableName]: { Keys },
-                },
-            })
-            .promise();
-
-        const metadataItems = items.Responses?.[Const.TileMetadata.TableName];
-        if (metadataItems == null) throw new Error('Failed to fetch tile metadata');
-
-        // TODO diff the arrays see what is missing?
-        if (metadataItems.length < Keys.length) throw new Error('Missing fetched items');
-
-        const results = metadataItems.map((c) => DynamoDB.Converter.unmarshall(c)) as TileMetadataImageryRecord[];
-
-        for (const result of results) this.imagery.set(result.id, result);
+        let mappedKeys = keys.map(toId);
+        let results: TileMetadataImageryRecord[] | null = null;
+
+        while (mappedKeys.length != 0) {
```

Comment:
call it paranoia, but I never like direct equals comparisions on while loops, no idea how .length could get below zero but if it does it will go forever!

Reply:
I call it paranoia ;)